libbindgen/src/ir/comp.rs

```diff
@@ -506,6 +509,7 @@ impl CompInfo {
         debug!("CompInfo::from_ty({:?}, {:?})", kind, cursor);
 
         let mut ci = CompInfo::new(kind);
+        ci.is_declaration = cursor.definition().unwrap().is_declaration();
```

Comment:
what if we modify the `from_ty` function to accept a cursor as well? We could simple use `clang_isDeclaration()` then.